src/Analysis/Engine/Impl/Infrastructure/PathUtils.cs

```diff
@@ -382,6 +408,46 @@ static class PathUtils {
             return null;
         }
 
+        public static bool IsNormalizedPath(string path) {
```

Comment:
I wonder if we can just `PathUtil.NormalizePath(path) == path`?

Reply:
`PathUtil.NormalizePath(path)` produces enormous amount of garbage. During initial load, some of that garbage even ends up in Gen2. So the goal was the opposite - don't create new path if the original one is normalized already.